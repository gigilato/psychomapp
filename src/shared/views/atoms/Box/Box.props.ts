import { ViewProps } from 'react-native'
import { AnimateProps } from 'react-native-reanimated'

export type IBoxProps = ViewProps
export type IAnimatedBoxProps = AnimateProps<IBoxProps>
