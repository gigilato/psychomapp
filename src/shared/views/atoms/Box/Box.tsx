import clsx from 'clsx'
import { Component } from 'react'
import { View } from 'react-native'
import Animated from 'react-native-reanimated'

import { IBoxProps } from '$views/atoms/Box/Box.props'

export const Box = View
export const AnimatedBox = Animated.createAnimatedComponent(Box)

export const HStack = ({ className, ...props }: IBoxProps) => (
  <View className={clsx('flex-row', className)} {...props} />
)
class HStackComponent extends Component<IBoxProps> {
  render() {
    return <Center {...this.props} />
  }
}
export const AnimatedHStack = Animated.createAnimatedComponent(HStackComponent)

export const Center = ({ className, ...props }: IBoxProps) => (
  <View className={clsx('justify-center items-center', className)} {...props} />
)
class CenterComponent extends Component<IBoxProps> {
  render() {
    return <Center {...this.props} />
  }
}
export const AnimatedCenter = Animated.createAnimatedComponent(CenterComponent)

export const Screen = ({ className, ...props }: IBoxProps) => (
  <View className={clsx('flex-1', className)} {...props} />
)
