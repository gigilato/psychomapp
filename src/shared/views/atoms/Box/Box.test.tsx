import { render } from '@testing-library/react-native'

import { Box, HStack, Center, Screen } from './Box'

const Component = () => (
  <Box className="flex-1 bg-orange justify-center">
    <HStack className="bg-grey-classic">
      <Box className="h-[80] w-[80] bg-success-tint200" />
      <Box className="p-[60] mx-s bg-error" />
      <Box className="h-[20] w-xxl bg-primary-electric" />
    </HStack>
  </Box>
)

describe('Box', () => {
  it('renders correctly', async () => {
    const component = render(<Component />)
    expect(component).toMatchSnapshot()
  })
})

describe('Center', () => {
  it('renders correctly', async () => {
    const component = render(<Center />)
    expect(component).toMatchSnapshot()
  })
})

describe('Screen', () => {
  it('renders correctly', async () => {
    const component = render(<Screen />)
    expect(component).toMatchSnapshot()
  })
})
