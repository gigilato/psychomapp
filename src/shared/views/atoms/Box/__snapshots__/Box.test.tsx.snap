// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box renders correctly 1`] = `
<View>
  <View>
    <View />
    <View />
    <View />
  </View>
</View>
`;

exports[`Center renders correctly 1`] = `<View />`;

exports[`Screen renders correctly 1`] = `<View />`;
